### PR TITLE
Add merging principles documentation

### PR DESCRIPTION
**Pull request by patrickkwang**

### CODE REVIEW

File: principles/merging.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+### Knowledge graph edges
+
+Knowledge graph edge values should be merged by:
+
+* concatenating their `attributes` lists
+
+## Results
+
+Results should be concatenated.
```
Comment (finnagin) [inline principles/merging.md:43]: Should there be some sort of merging mentioned here for results that are considered equivalent? 

In the equivalence.md file we mention that results are equivalent "when node bindings and edge bindings are identical, after accounting for knowledge graph node key equivalence". 

Would this lead to a merging of these specific results or do we want both equivalent results included as separate entries?

**Review by edeutsch — APPROVED** [review]

File: principles/equivalence.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Equivalence
+
+The following defines when TRAPI components should be considered equivalent.
+
+* **query graph nodes** - when keys and values are identical
```
Comment (cbizon) [inline principles/equivalence.md:5]: All keys and values?   What if it is a bound node, then should it be similar to KG nodes, i.e. equivalent identifiers?

File: principles/equivalence.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+* **query graph edges** - when keys and values are identical
+* **knowledge graph nodes** - when keys are equivalent per node-norm
+* **knowledge graph edges** - when subjects, predicates, objects, and relations are identical
```
Comment (cbizon) [inline principles/equivalence.md:8]: Accounting for nodenorm?

File: principles/equivalence.md (hunk 2 above)
Comment (cbizon) [reply principles/equivalence.md:5]: Name too?

File: principles/merging.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+Knowledge graph nodes should be merged if and only if their keys are identical.
+
+Knowledge graph edges should be merged if and only if they share a `subject`, `predicate`, `object`, and `relation`. If `relation` is provided for one edge and omitted by the other, they should not be merged.
+
+### Knowledge graph nodes
+
+Knowledge graph node values should be merged by:
+
+* finding the union of their `categories` lists
+* concatenating their `attributes` lists
```
Comment (cbizon) [inline principles/merging.md:33]: Do we want any attribute merging?

**Review by finnagin — APPROVED** [review]

**Review by dkoslicki — APPROVED** [review]

**Comment by dkoslicki** [comment]
@patrickkwang Is this good to merge? I noticed it was marked as draft (I just changed it to `ready for review`

**Pull request merged**